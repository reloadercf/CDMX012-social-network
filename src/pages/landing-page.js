// eslint-disable-next-line import/no-cycle
import { onNavigate } from '../app.js';

export const landingPage = () => {
  const lpContent = document.createElement('div');
  lpContent.setAttribute('class', 'lp-content');

  const lpContainer = document.createElement('div');
  lpContainer.setAttribute('class', 'landing-page-container');

  // header section
  const header = document.createElement('header');
  header.setAttribute('class', 'logo-header');

  const logo = document.createElement('img');
  logo.setAttribute('class', 'logo-book');
  logo.setAttribute('src', './assets/logo sin fondo 1.png');
  logo.setAttribute('alt', 'logo book reads');

  const bookreads = document.createElement('img');
  bookreads.setAttribute('class', 'titleBookReads');
  bookreads.setAttribute('src', './assets/nombre_sin_fondo-removebg-preview 1.png');
  bookreads.setAttribute('alt', 'titleBookReads');

  const slogan = document.createElement('h1');
  slogan.innerHTML = 'Read with me';
  slogan.setAttribute('class', 'sloganSN');

  header.append(logo, bookreads, slogan);

  // buttons container
  const buttonsContainer = document.createElement('div');
  buttonsContainer.setAttribute('class', 'sign-up-in-container');

  // button sign-up section
  const signUpBtn = document.createElement('input');
  signUpBtn.setAttribute('type', 'button');
  signUpBtn.setAttribute('class', 'btn-signUp');
  signUpBtn.setAttribute('id', 'btn-signUp-LP');
  signUpBtn.setAttribute('value', 'Sign Up');

  signUpBtn.addEventListener('click', (event) => {
    event.preventDefault();
    onNavigate('/sign-up');
  });

  // button sign-in section
  const signInBtn = document.createElement('input');
  signInBtn.setAttribute('type', 'button');
  signInBtn.setAttribute('class', 'btn-signIn');
  signInBtn.setAttribute('id', 'btn-signIn-LP');
  signInBtn.setAttribute('value', 'Sign In');

  signInBtn.addEventListener('click', (event) => {
    event.preventDefault();
    onNavigate('/sign-in');
  });

  buttonsContainer.append(signUpBtn, signInBtn);

  lpContainer.append(header, buttonsContainer);

  const lpImage = document.createElement('div');
  lpImage.setAttribute('class', 'landing-page-img');

  lpContent.append(lpContainer, lpImage);

  return lpContent;
};
