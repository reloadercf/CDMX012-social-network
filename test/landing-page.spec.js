/**
 * @jest-environment jsdom
 */
import { landingPage } from '../src/pages/landing-page.js';

jest.mock('../src/firebase-imports.js');

describe('Tests  of landingPage', () => {
  test('render landingpage', () => {
    const renderLanding = landingPage();
    document.body.innerHTML = '<div id="root"></div>';
    const rootDiv = document.getElementById('root');
    rootDiv.appendChild(renderLanding);
    expect(rootDiv.innerHTML).toMatchSnapshot();
  });
});
