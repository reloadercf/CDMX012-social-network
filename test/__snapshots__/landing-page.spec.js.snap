// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests  of landingPage render landingpage 1`] = `"<div class=\\"lp-content\\"><div class=\\"landing-page-container\\"><header class=\\"logo-header\\"><img class=\\"logo-book\\" src=\\"./assets/logo sin fondo 1.png\\" alt=\\"logo book reads\\"><img class=\\"titleBookReads\\" src=\\"./assets/nombre_sin_fondo-removebg-preview 1.png\\" alt=\\"titleBookReads\\"><h1 class=\\"sloganSN\\">Read with me</h1></header><div class=\\"sign-up-in-container\\"><input type=\\"button\\" class=\\"btn-signUp\\" id=\\"btn-signUp-LP\\" value=\\"Sign Up\\"><input type=\\"button\\" class=\\"btn-signIn\\" id=\\"btn-signIn-LP\\" value=\\"Sign In\\"></div></div><div class=\\"landing-page-img\\"></div></div>"`;
